protocol/0082-RFPR-on_chain_referral_program.md

```diff
@@ -0,0 +1,235 @@
+                "referral_discount_factor": 0.002,
+            }
+        ],
+        start_epoch: 107
+        end_epoch: 114,
+        window_length: 1,
+    }
+}
+```
+In addition to the conditions which apply when creating a referral program, the following apply:
+- a proposer cannot update a program if its state is `STATUS_REJECTED`, `STATUS_STOPPED`, or `STATUS_CLOSED`.
+- a proposer cannot update the `name` of a program.
+- a proposer cannot update the `start_epoch` if the program is `STATUS_ACTIVE`
+
+## Referral Program Lifecycle:
+
+After a referral program [proposal](#governance-proposals) is validated and accepted by the network, a referral program is created and can be one of the following states. The current state of a program should be exposed via an API.
+
+| Status             | Benefits Enabled | Condition for entry                                       | Condition for exit                                   |
+| ------------------ | ---------------- | --------------------------------------------------------- | ---------------------------------------------------- |
+| `STATUS_PROPOSED`  | No               | Governance proposal valid and accepted                    | Governance proposal voting period ends               |
+| `STATUS_REJECTED`  | No               | Governance vote fails                                     | na                                                   |
+| `STATUS_PENDING`   | No               | Governance vote passes                                    | Current epoch greater than or equal to `start_epoch` |
+| `STATUS_STOPPED`   | No               | There is already an existing program with `STATUS_ACTIVE` | na                                                   |
+| `STATUS_ACTIVE`    | Yes              | Previously `STATUS_PENDING`                               | Current epoch less than or equal to `end_epoch`      |
+| `STATUS_CLOSED`    | No               | Previously `STATUS_ACTIVE`                                | na                                                   |              
+
+## Team Mechanics
+A team is comprised of a referrer and all their referees. There can only ever be one referrer per team but the number of referees is unlimited.
+
+### Creating a team
+
+To create a team and generate a referral code, a party must fulfil the following criteria:
+
+- party must not currently be a **referrer**
+- party must not currently be a **referee**
+- party must be staking at least `referralProgram.minStakedVegaTokens` tokens.
+
+This staking requirement is constant. If a referrer un-stakes enough tokens to fall below the requirement, they and their referees will no long be eligible for program benefits. If the referrer re-stakes enough tokens to fulfil the staking requirement, the team will become eligible for referral program benefits.
+
+A separate URL hosted on IPFS should allow a party to connect their Vega wallet and fill in the following optional team information:
```

Comment:
Is this a front end requirement? In that case that can be left as an aside but we should formulate what the node has to provide in terms of transactions it accepts to facilitate this. 

And perhaps it's best not to talk about the front-end part in the protocol part of the spec.

Reply:
Yeah I vote kill anything non-protocol related (even datanodey things make me sad)

Reply:
Yeah I'd remove front end requirements, we just need the transactions we define and data node APIs to facilitate the front end, in this spec.

There should be a front end spec/ticket (check with FE team what they need) as well for this though.